src/librustc_lexer/src/lib.rs

```diff
@@ -235,16 +235,15 @@ pub enum Base {
 /// Line won't be skipped if it represents a valid Rust syntax
 /// (e.g. "#![deny(missing_docs)]").
 pub fn strip_shebang(input: &str) -> Option<usize> {
-    debug_assert!(!input.is_empty());
-    let s: &str = &remove_whitespace(input);
-    if !s.starts_with("#!") || s.starts_with("#![") {
-        return None;
+    if let Some(first_line) = input.lines().next() {
+        let prefix = first_line.chars().filter(|c| !is_whitespace(*c)).collect::<Vec<_>>();
```

Comment:
```suggestion
        let prefix = first_line.chars().filter(|c| !is_whitespace(*c)).take(3).collect::<Vec<_>>();
```

Reply:
You could also turn `if let Some(first_line) = input.lines().next() { ... }` into `input.lines().next().and_then(|first_line| ...)` here.